src/TypeSpec.Extension/Emitter.Csharp/src/lib/converter.ts

```diff
@@ -333,21 +333,19 @@ function fromUnionType(
     context: SdkContext,
     models: Map<string, InputModelType>,
     enums: Map<string, InputEnumType>
-): InputUnionType | InputType {
+): InputUnionType {
     const itemTypes: InputType[] = [];
     for (const value of union.values) {
-        const inputType = fromSdkType(value, context, models, enums);
-        itemTypes.push(inputType);
+        const valueType = fromSdkType(value, context, models, enums);
```

Comment:
It looks when we update the emitter code for type consolidation, there is no test change needed, so it shows our test coverage for emitter is not enough.
Not the requirement for this PR, but please consider to provide more unit tests for emitter as we continue this consolidation work.

Reply:
There is an issue that is tracking to reduce the size of emitter tests: https://github.com/microsoft/typespec/issues/3337
Therefore I would like to see how it goes before we add new tests